kuma/core/i18n.py

```diff
@@ -1,4 +1,165 @@
+                return django_language_code_to_kuma(code)
+        # If fr-fr is not supported, try fr-ca.
+        for supported_code in supported_lang_codes:
+            if supported_code.startswith(generic_lang_code + '-'):
+                return django_language_code_to_kuma(supported_code)
+    raise LookupError(raw_lang_code)
+
+
+def get_language_from_path(path):
+    """
+    Returns the language-code if there is a valid language-code
+    found in the `path`.
+
+    If `strict` is False (the default), the function will look for an alternative
+    country-specific variant when the currently checked is not found.
+
+    Based on Django 1.8.19's get_language_from_path from
+    django/utils/translation/trans_real.py, with changes:
+
+    * Don't accept or pass strict parameter (assume strict=False).
+    """
+    regex_match = language_code_prefix_re.match(path)
+    if not regex_match:
+        return None
+    lang_code = regex_match.group(1)
+    try:
+        return get_supported_language_variant(lang_code)
+    except LookupError:
+        return None
+
+
+def get_language_from_request(request):
+    """
+    Analyzes the request to find what language the user wants the system to
+    show. Only languages listed in settings.LANGUAGES are taken into account.
+    If the user requests a sublanguage where we have a main language, we send
+    out the main language.
+
+    If check_path is True, the URL path prefix will be checked for a language
+    code, otherwise this is skipped for backwards compatibility.
+
+    Based on Django 1.8.19's get_language_from_request from
+    django/utils/translation/trans_real.py, with changes:
+
+    * Always check the path (check_path=False)
+    * Don't check session language
+    """
+    lang_code = get_language_from_path(request.path_info)
+    if lang_code is not None:
+        return lang_code
+
+    lang_code = request.COOKIES.get(settings.LANGUAGE_COOKIE_NAME)
+
+    try:
+        return get_supported_language_variant(lang_code)
+    except LookupError:
+        pass
+
+    accept = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
+    for accept_lang, unused in parse_accept_lang_header(accept):
+        if accept_lang == '*':
+            break
+
+        if not language_code_re.search(accept_lang):  # pragma: no cover
+            # Check added with a security fix:
+            # https://www.djangoproject.com/weblog/2007/oct/26/security-fix/
+            # It is unclear how to trigger this branch, so skipping coverage.
+            continue
+
+        try:
+            return get_supported_language_variant(accept_lang)
+        except LookupError:
+            continue
+
+    # Fallback to default settings.LANGUAGE_CODE.
+    # Django supports a case when LANGUAGE_CODE is not in LANGUAGES
+    # (see https://github.com/django/django/pull/824). but our LANGUAGE_CODE is
+    # always in LANGUAGES.
+    assert settings.LANGUAGE_CODE == settings.LANGUAGES[0][0]
```

Comment:
I dont know if we should check it in every request. maybe add a test where its checked?

Reply:
It's indirectly checked here:

https://github.com/mozilla/kuma/blob/cdfce6d69f6740d2061c8795cbfc7b12ba05b92d/kuma/settings/common.py#L236-L237

I think it makes sense to add a test and remove the assertion here.